src/Symfony/Component/Translation/Provider/ProvidersFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Provider;
+
+use Symfony\Component\Translation\Exception\UnsupportedSchemeException;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ *
+ * @experimental in 5.3
+ */
+class ProvidersFactory
```

Comment:
Could you help me understand why we need this class. It will call the provider factory...

It looks like we have a factory to call a factory.

Reply:
It was designed by @odolbeau at the very beginning of this PR, may be he could help us?

Reply:
@Nyholm According to [your suggestion](https://github.com/symfony/symfony/pull/38475#discussion_r582643936), this class should also be renamed to `TranslationProviderCollectionFactory` a name which is clearer I think.
The goal is to build the collection of providers according to the configuration provided.